### PR TITLE
Update FAve 84H RGB firmware

### PR DESCRIPTION
**Pull request by ToastyStoemp**
## Description

Added VIA compatible RGB Matrix options

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/linworks/fave84h/config.h [hunk 1]
```diff
@@ -90,23 +92,34 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
...
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#define ENABLE_RGB_MATRIX_SPLASH
+#define ENABLE_RGB_MATRIX_MULTISPLASH
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+
 #endif
+
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (zvecr) [inline keyboards/linworks/fave84h/config.h:125]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Doesnt exist any more.

**Comment by drashna** [comment]
Mostly, this is adding support for via, for rgb light, correct?   

That's what it looks like.   But develop includes these changes already.  Compiling from develop will handle this already. !

**Comment by ToastyStoemp** [comment]
> Mostly, this is adding support for via, for rgb matrix, correct?
> 
> That's what it looks like. But develop includes these changes already. Compiling from develop will handle this already. !

Sorry for the late reply, yes (did reduce the max brightness a little also) but this is pretty much just the VIA hack implementation

**Review by drashna — APPROVED** [review]

File: keyboards/linworks/fave84h/rules.mk [hunk 2]
```diff
@@ -20,4 +20,4 @@ BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
 RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
 AUDIO_ENABLE = no           # Audio output
 RGB_MATRIX_ENABLE = yes
-RGB_MATRIX_DRIVER = WS2812
+RGB_MATRIX_DRIVER = WS2812
```
Comment (fauxpark) [inline keyboards/linworks/fave84h/rules.mk:23]:
```suggestion
RGB_MATRIX_DRIVER = WS2812
```
All lines should end in a newline character.

File: keyboards/linworks/fave84h/keymaps/via/rules.mk [hunk 3]
```diff
@@ -1 +1,2 @@
 VIA_ENABLE = yes
+LTO_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/linworks/fave84h/keymaps/via/rules.mk:2]:
```suggestion
LTO_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by ToastyStoemp** [comment]
bump

**Comment by drashna** [comment]
Thanks!

**Pull request merged**